app/bundles/CampaignBundle/Executioner/EventExecutioner.php

```diff
@@ -122,12 +123,14 @@ public function executeForContact(Event $event, Lead $contact, Responses $respon
     }
 
     /**
+     * @param Collection<int|string, mixed>|ArrayCollection $events
+     *
      * @throws Dispatcher\Exception\LogNotProcessedException
      * @throws Dispatcher\Exception\LogPassedAndFailedException
      * @throws Exception\CannotProcessEventException
      * @throws Scheduler\Exception\NotSchedulableException
      */
-    public function executeEventsForContact(ArrayCollection $events, Lead $contact, Responses $responses = null, Counter $counter = null)
+    public function executeEventsForContact(Collection $events, Lead $contact, Responses $responses = null, Counter $counter = null)
```

Comment:
Changing signature of public and protected methods is considered a BC break. Is this necessary? It's fine to do so for private methods.

Reply:
This was only necessary to fix problems with phpstan. I thought it was safe because ArrayCollection implements Collection. 

Almost issues were related to function signatures. If it is considered a BC break,  I will need to restore any changes related to it. And I will need help to fix phpstan issues. Because in my opinion there is a real problem with the types.

Reply:
You are right. `Collection` should not cause any BC breaks.